src/Agent.Worker/ExpressionManager.cs

```diff
@@ -0,0 +1,101 @@
+using Microsoft.TeamFoundation.DistributedTask.WebApi;
+using Microsoft.VisualStudio.Services.Agent.Util;
+using DT = Microsoft.VisualStudio.Services.DistributedTask.Expressions;
+
+namespace Microsoft.VisualStudio.Services.Agent.Worker
+{
+    [ServiceLocator(Default = typeof(ExpressionManager))]
+    public interface IExpressionManager : IAgentService
+    {
+        bool Evaluate(IExecutionContext context, string condition);
+    }
+
+    public sealed class ExpressionManager : AgentService, IExpressionManager
+    {
+        public bool Evaluate(IExecutionContext executionContext, string condition)
+        {
+            ArgUtil.NotNull(executionContext, nameof(executionContext));
+
+            // Parse the condition.
+            var expressionTrace = new TraceWriter(executionContext);
+            var parser = new DT.Parser();
+            var extensions = new DT.IExtensionInfo[]
+            {
+                new DT.ExtensionInfo<AlwaysNode>(name: Constants.Expressions.Always, minParameters: 0, maxParameters: 0),
+                new DT.ExtensionInfo<SucceededNode>(name: Constants.Expressions.Succeeded, minParameters: 0, maxParameters: 0),
+                new DT.ExtensionInfo<SucceededOrFailedNode>(name: Constants.Expressions.SucceededOrFailed, minParameters: 0, maxParameters: 0),
+                new DT.ExtensionInfo<VariablesNode>(name: Constants.Expressions.Variables, minParameters: 1, maxParameters: 1),
+            };
+            DT.Node tree = parser.CreateTree(condition, expressionTrace, extensions) ?? new SucceededNode();
+
+            // Evaluate the tree.
+            var evaluationContext = new DT.EvaluationContext(expressionTrace, state: executionContext);
+            return tree.Evaluate(evaluationContext).ConvertToBoolean(evaluationContext);
+        }
+
+        public sealed class TraceWriter : DT.ITraceWriter
+        {
+            private readonly IExecutionContext _executionContext;
+
+            public TraceWriter(IExecutionContext executionContext)
+            {
+                ArgUtil.NotNull(executionContext, nameof(executionContext));
+                _executionContext = executionContext;
+            }
+
+            public void Info(string message)
+            {
+                _executionContext.Output(message);
+            }
+
+            public void Verbose(string message)
```

Comment:
I need to understand more about how this will be consumed on the server side. On the agent side, I want everything to go to `##debug`. Currently I don't write anything to Info. I'm unsure right now whether we will have any localized content to log as info.